sycl/include/CL/sycl/buffer.hpp

```diff
@@ -45,8 +44,10 @@ template <typename T, int dimensions = 1,
           typename __Enabled = typename detail::enable_if_t<(dimensions > 0) &&
                                                             (dimensions <= 3)>>
 class buffer {
+  // TODO check is_device_copyable<T>::value after converting sycl::vec into a
+  // trivially copyable class.
   static_assert(
-      is_device_copyable<T>::value,
+      !std::is_same<T, std::string>::value,
       "The underlying data type of a buffer 'T' must be device copyable");
```

Comment:
It is better to change wording to make it more specific.

Reply:
Well, I've made the wording more specific.